profiler/main.py

```diff
@@ -0,0 +1,169 @@
+import argparse
+import os
+import re
+import subprocess
+from datetime import date, datetime
+from subprocess import PIPE
+from typing import Optional
+
+import somacore
+
+import tiledbsoma
+from data import FileBasedProfileDB, ProfileData, ProfileDB
+
+TILEDB_STATS_FILE_PATH = "./tiledb_stats.txt"
+
+
+def parse_time_output(
+    output: str, process: str, prof1: Optional[str], prof2: Optional[str]
+):
+    """Parse the time utility output to extracting performance and memory metrics"""
+    lines = output.split("\n")
+    lines = lines[-20:]
+    for idx, line in enumerate(lines):
+        perf_match = re.match(
+            "\s+(\d+\.\d+) real\s+(\d+\.\d+) user\s+(\d+\.\d+) sys\s*", line
+        )
+        if perf_match:
+            real_time = float(perf_match.groups()[0])
+            usr_time = float(perf_match.groups()[1])
+            sys_time = float(perf_match.groups()[2])
+            print(
+                f"real_time = {real_time} user_time = {usr_time} sys_time = {sys_time}"
+            )
+        max_resident_set_size_match = re.match(
+            "\s+(\d+)\s+maximum resident set size\s*", line
+        )
+        if max_resident_set_size_match:
+            max_resident_set_size = int(max_resident_set_size_match.groups()[0])
+            print(f"max_resident_set_size = {max_resident_set_size}")
+        page_reclaims_match = re.match("\s+(\d+)\s+page reclaims\s*", line)
+        if page_reclaims_match:
+            page_reclaims = int(page_reclaims_match.groups()[0])
+            print(f"page_reclaims = {page_reclaims}")
+        page_faults_match = re.match("\s+(\d+)\s+page faults\s*", line)
+        if page_faults_match:
+            page_faults = int(page_faults_match.groups()[0])
+            print(f"page_faults = {page_faults}")
+        cycles_elapsed_match = re.match("\s+(\d+)\s+cycles elapsed\s*", line)
+        if cycles_elapsed_match:
+            cycles_elapsed = int(cycles_elapsed_match.groups()[0])
+            print(f"cycles_elapsed = {cycles_elapsed}")
+        peak_memory_footprint_match = re.match(
+            "\s+(\d+)\s+peak memory footprint\s*", line
+        )
+        if peak_memory_footprint_match:
+            peak_memory_footprint = int(peak_memory_footprint_match.groups()[0])
+            print(f"peak_memory_footprint = {peak_memory_footprint}")
+    tiledb_stats: str = None
+    if os.path.isfile(TILEDB_STATS_FILE_PATH):
+        with open(TILEDB_STATS_FILE_PATH, "r") as f:
+            print("TileDB stats found")
+            tiledb_stats = f.read()
+    custom_out = [prof1, prof2]
+    data: ProfileData = ProfileData(
+        process=" ".join(process),
```

Comment:
what is the purpose of the `join` call?

Reply:
Process here is a list of strings. The type hint is wrong. :(